src/save_options/registration.py

```diff
@@ -0,0 +1,111 @@
+import contextlib
+import inspect
+from collections.abc import Callable, Sequence
+from typing import Any
+
+from unrealsdk import logging
+
+from mods_base import BaseOption, GroupedOption, Mod, NestedOption
+from save_options.options import SaveOption
+
+# Allowing GroupedOption and NestedOption if all their children are SaveOption
+ModSaveOptions = dict[str, BaseOption]
+registered_save_options: dict[str, ModSaveOptions] = {}
+
+SaveCallback = Callable[[], None]
+save_callbacks: dict[str, SaveCallback] = {}
+
+LoadCallback = Callable[[], None]
+load_callbacks: dict[str, LoadCallback] = {}
+
+registered_mods: dict[str, Mod] = {}
+
+
+def _treat_as_save_option(obj: Any) -> bool:
+    if isinstance(obj, SaveOption):
+        return True
+    if isinstance(obj, GroupedOption | NestedOption):
+        if all(_treat_as_save_option(child) for child in obj.children):
+            return True
+        logging.error(
+            f"Option {obj.identifier} has both regular BaseOption and SaveOption"
+            f"defined as children. SaveOption instances will be ignored"  # noqa: COM812
+        )
+    return False
+
+
+def register_save_options(  # noqa: C901, D417
+    mod: Mod,
+    *,
+    save_options: Sequence[BaseOption] | None = None,
+    on_save: Callable[[], None] | None = None,
+    on_load: Callable[[], None] | None = None,
+    mod_identifier: str | None = None,
+) -> None:
+    """
+    Registers save options - values stored on the save options are saved to the character save file.
+
+    Positional arg:
+        mod: The Mod instance.
+
+    Keyword Args:
+        save_options: A sequence of SaveOption instances to register. If None, variables in the
+                      calling module's scope are searched, and any SaveOption found is appended.
+        on_save: A callback to run any time the game is saved. Intended use is to update any
+                 SaveOption values before they are written to the save file. If None, we search the
+                calling module's scope for "on_save".
+        on_load: A callback to run right after the player's save data is applied to the player upon
+                 entering the game. Intended use is to apply any custom save option values to the
+                 player. If None, we search the calling module's scope for "on_load".
+        mod_identifier: A string to identify the mod in the save file. If None, module.__name__ is
+                        used.
```

Comment:
I think this could do with an ascii art table like in `build_mod`.

One of the lines (if you keep it) needs an extra space.